### ISSUE TITLE
again: error: use of undeclared identifier 'unused'

### ISSUE THREAD

**Issue by cornellouis** [tmux/tmux#1801]
### Issue description

I could not successfully complete the installation instructions.  Particularly the make command.

 tmux-2.0.tar.gz
`
tmux 2.9a
 make -v
GNU Make 3.81
MacOS 10.14.5
<~/tmux-2.0>-> gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/c++/4.2.1
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
<~/tmux-2.0>-> clang -v
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

* tmux version (`tmux -V`).
* Platform (`uname -sp`).
* $TERM inside and outside of tmux (`echo $TERM`).
* Logs from tmux (`tmux kill-server; tmux -vv new`).

Darwin i386

xterm-256color
`

Same issue as https://github.com/tmux/tmux/issues/1562

But I don't know why that issue was closed.  I didn't find any resolution in the comments.

`depbase=`echo compat/closefrom.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
	gcc -DPACKAGE_NAME=\"tmux\" -DPACKAGE_TARNAME=\"tmux\" -DPACKAGE_VERSION=\"2.0\" -DPACKAGE_STRING=\"tmux\ 2.0\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"tmux\" -DVERSION=\"2.0\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_BITSTRING_H=1 -DHAVE_CURSES_H=1 -DHAVE_DIRENT_H=1 -DHAVE_FCNTL_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_NCURSES_H=1 -DHAVE_PATHS_H=1 -DHAVE_STDINT_H=1 -DHAVE_SYS_DIR_H=1 -DHAVE_TERM_H=1 -DHAVE_UTIL_H=1 -DHAVE_BZERO=1 -DHAVE_DIRFD=1 -DHAVE_FLOCK=1 -DHAVE_SYSCONF=1 -DHAVE_CFMAKERAW=1 -DHAVE_B64_NTOP=1 -DHAVE_FORKPTY=1 -DHAVE_DAEMON=1 -DHAVE_SETENV=1 -DHAVE_STRLCPY=1 -DHAVE_STRLCAT=1 -DHAVE_ASPRINTF=1 -DHAVE_FGETLN=1 -DHAVE_FPARSELN=1 -DHAVE_STRCASESTR=1 -DHAVE_STRSEP=1 -DHAVE_CFMAKERAW=1 -DHAVE_OPENAT=1 -DHAVE_DECL_OPTARG=1 -DHAVE_DECL_OPTIND=1 -DHAVE_DECL_OPTRESET=1 -DHAVE_GETOPT=1 -DHAVE_BSD_TYPES=1 -DHAVE___PROGNAME=1 -DBROKEN_CMSG_FIRSTHDR=1 -I.   -I/usr/local/Cellar/libevent/2.1.10/include -DTMUX_CONF="\"/etc/tmux.conf\""  -iquote.      -std=gnu99 -O2    -MT compat/closefrom.o -MD -MP -MF $depbase.Tpo -c -o compat/closefrom.o compat/closefrom.c &&\
	mv -f $depbase.Tpo $depbase.Po
In file included from compat/closefrom.c:34:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/dirent.h:80:2: error: use of undeclared identifier 'unused'
        __unused long   __padding; /* (__dd_rewind space left for bincompat) */
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/sys/cdefs.h:161:40: note: expanded from macro '__unused'
#define __unused        __attribute__((unused))
                                       ^
./tmux.h:67:32: note: expanded from macro 'unused'
#define unused __attribute__ ((unused))
                               ^
1 error generated.
make: *** [compat/closefrom.o] Error 1`

Please read https://github.com/tmux/tmux/blob/master/.github/CONTRIBUTING.md
before opening an issue.

**Comment by nicm** [comment]
Seems like a problem with your toolchain.

**Comment by nicm** [comment]
Show config.log and the full build output please, without make -j please.

**Comment by cornellouis** [comment]
[config.log](https://github.com/tmux/tmux/files/3307179/config.log)
[make_output.txt](https://github.com/tmux/tmux/files/3307193/make_output.txt)

Sorry if I didn't generate these properly.  I don't know much about building...

**Comment by cornellouis** [comment]
I've tried doing this a number of times, but it hasn't helped:  https://donatstudios.com/MojaveMissingHeaderFiles

**Comment by nicm** [comment]
This is tmux 2.0 not tmux 2.9a. Does tmux 2.9a build?

**Comment by cornellouis** [comment]
I downloaded 2.9a and the build worked.  I just realized that this whole thing was about trying to get tmux-zen to work, and that git page had linked to tmux 2.0, I guess because it was outdated.  I already had 2.9a installed... so I'm an idiot.  I was just blindly following the install instructions.  I thought I was building something related to zen.  I guess it's not a priority to get 2.0 build working on Mojave.
